test/parallel_api/ranges/std_ranges_sort.pass.cpp

```diff
@@ -21,18 +21,24 @@ main()
 #if _ENABLE_STD_RANGES_TESTING
     using namespace test_std_ranges;
     namespace dpl_ranges = oneapi::dpl::ranges;
+    auto sort_checker = 
+#if _ONEDPL_STD_RANGES_ALGO_CPP_FUN
```

Comment:
Could you introduce a similar macro in the tests in order not to rely on oneDPL internals?

Reply:
I still strong disagree to code duplication.
Rather lets consider a an option to make public that macro....

Reply:
I do not think that having a public macro for that is a good idea given that it is a workaround for a niche issue, not resulting in enablement of any oneDPL feature. 

I guess we need a mediator here.

Reply:
Relocating my comment here, I missed this discussion in my initial review.  I agree with Dmitriy here, this has been the convention as established by @akukanov.

We have tried to stay away from using internal helpers / utilities / implementation details in our tests.  

If we want to use the same macro here, I think we probably want to re-define it here `test/support/utils.h`, or somewhere similar as `_TEST_PREPARE_CALLABLE`, or something like that, then use that throughout the tests.

Reply:
In the essence, here we have two good engineering practices contradicting each other, and we need to choose between "avoid code duplication" and "avoid test dependencies on implementation internals". I choose the second, because most if not all code duplications between the implementation and its tests are simple self-contained utilities or macros and duplicating those does not seem problematic to me.

Moreover, strictly speaking, the tests in this patch do not even have to use `_ONEDPL_PREPARE_CALLABLE` - range algorithms can always be wrapped into a lambda, possibly with a small "syntax sugar" helper.

Reply:
I see..
But we missed a way with passing a callable function by reference - `foo(f&& f)`  - https://godbolt.org/z/j1P6MKdvn
Is the implementation of a C++ function/algorithm with doesn't match to `foo(f&& f)`?